x-pack/plugins/cases/public/components/similar_cases/use_similar_cases_columns.tsx

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo } from 'react';
+import { css } from '@emotion/react';
+import type {
+  EuiTableActionsColumnType,
+  EuiTableComputedColumnType,
+  EuiTableFieldDataColumnType,
+} from '@elastic/eui';
+import { EuiBadgeGroup, EuiBadge, EuiHealth, EuiToolTip, RIGHT_ALIGNMENT } from '@elastic/eui';
+import { Status } from '@kbn/cases-components/src/status/status';
+
+import { CaseSeverity } from '../../../common/types/domain';
+import type { CaseUI, SimilarCaseUI } from '../../../common/ui/types';
+import { getEmptyCellValue } from '../empty_value';
+import { FormattedRelativePreferenceDate } from '../formatted_date';
+import { CaseDetailsLink } from '../links';
+import { TruncatedText } from '../truncated_text';
+import { severities } from '../severity/config';
+import { useCasesColumnsConfiguration } from '../all_cases/use_cases_columns_configuration';
+import * as i18n from './translations';
+
+type SimilarCasesColumns =
+  | EuiTableActionsColumnType<SimilarCaseUI>
+  | EuiTableComputedColumnType<SimilarCaseUI>
+  | EuiTableFieldDataColumnType<SimilarCaseUI>;
+
+const LINE_CLAMP = 3;
+const getLineClampedCss = css`
+  text-overflow: ellipsis;
+  display: -webkit-box;
+  -webkit-line-clamp: ${LINE_CLAMP};
+  -webkit-box-orient: vertical;
+  overflow: hidden;
+  word-break: normal;
+`;
+
+const renderStringField = (field: string, dataTestSubj: string) =>
+  field != null ? <span data-test-subj={dataTestSubj}>{field}</span> : getEmptyCellValue();
+
+const SIMILARITIES_FIELD = 'similarities' as const;
+
+export interface UseSimilarCasesColumnsReturnValue {
+  columns: SimilarCasesColumns[];
+  rowHeader: string;
+}
+
+export const useSimilarCasesColumns = (): UseSimilarCasesColumnsReturnValue => {
+  const casesColumnsConfig = useCasesColumnsConfiguration(false);
+  const columns: SimilarCasesColumns[] = useMemo(
+    () => [
+      {
+        field: casesColumnsConfig.title.field,
+        name: casesColumnsConfig.title.name,
+        sortable: true,
+        render: (title: string, theCase: SimilarCaseUI) => {
+          if (theCase.id != null && theCase.title != null) {
+            const caseDetailsLinkComponent = (
+              <CaseDetailsLink detailName={theCase.id} title={theCase.title}>
+                <TruncatedText text={theCase.title} />
+              </CaseDetailsLink>
+            );
+
+            return caseDetailsLinkComponent;
+          }
+          return getEmptyCellValue();
+        },
+        width: '20%',
+      },
+      {
+        field: SIMILARITIES_FIELD,
+        name: i18n.SIMILARITY_REASON,
+        sortable: true,
+        render: (similarities: SimilarCaseUI['similarities'], theCase: SimilarCaseUI) => {
```

Comment:
We should introduce the `getLineClampedCss` to this column too because it can grow indefinitely and the UI doesn't look great like that.


<img width="589" alt="Screenshot 2024-11-14 at 19 05 38" src="https://github.com/user-attachments/assets/95ca7a3c-f117-410f-a29d-b39b5c08bab4">